dashboard/app.py

```diff
@@ -0,0 +1,670 @@
+
+        release_dates (list[datetime]): A list with release dates that the user selected
+
+        review_dates (list[datetime]): A list with review dates that the user selected
+
+        genres (list[str]): A list with game genres that the user selected
+
+        platforms (list[str]): A list with selected platforms
+
+    Returns:
+        Chart: A chart displaying plotted data
+    """
+    if len(titles) != 0:
+        df = df[df["title"].isin(titles)]
+    if len(release_dates) != 0:
+        df = df[df["release_date"].dt.floor("D").isin(release_dates)]
+    if len(review_dates) != 0:
+        df = df[df["review_date"].dt.floor("D").isin(review_dates)]
+    if len(genres) != 0:
+        df = df[df["genre"].isin(genres)]
+    df = df[df[platforms].any(axis=1)]
+    df = df[(df['sentiment'] >= minimum_sentiment) &
+            (df['sentiment'] <= maximum_sentiment)]
+
+    df = df.groupby(
+        "release_date").size().reset_index()
+    df.columns = ["release_date", "num_of_reviews"]
+    custom_ticks = [i for i in range(
+        0, df["num_of_reviews"].max() + 1)]
+
+    chart = alt.Chart(df).mark_line(
+        color="#44bd4f"
+    ).encode(
+        x=alt.X("release_date:O", title="Release Date",
+                timeUnit="yearmonthdate"),
+        y=alt.Y("num_of_reviews:Q", title="Number of Reviews", axis=alt.Axis(
+            values=custom_ticks, tickMinStep=1, titlePadding=10)),
+    ).properties(
+        title="New Reviews per Day",
+        width=800,
+        height=400
+    )
+
+    return chart
+
+
+def plot_reviews_per_game_frequency(df: DataFrame, titles: list[str], release_dates: list[datetime], review_dates: list[datetime],
+                                    genres: list[str], platforms: list[str], minimum_sentiment: float, maximum_sentiment: float) -> Chart:
+    """
+    Create a bar chart for the number of reviews per game
+
+    Args:
+        df (DataFrame): A pandas DataFrame containing all relevant game data
+
+        titles (list[str]):  A list with game titles that the user selected
+
+        release_dates (list[datetime]): A list with release dates that the user selected
+
+        review_dates (list[datetime]): A list with review dates that the user selected
+
+
+        genres (list[str]): A list with game genres that the user selected
+
+        platforms (list[str]): A list with selected platforms
+
+    Returns:
+        Chart: A chart displaying plotted data
+    """
+    if len(titles) != 0:
+        df = df[df["title"].isin(titles)]
+    if len(release_dates) != 0:
+        df = df[df["release_date"].dt.floor("D").isin(release_dates)]
+    if len(review_dates) != 0:
+        df = df[df["review_date"].dt.floor("D").isin(review_dates)]
+    if len(genres) != 0:
+        df = df[df["genre"].isin(genres)]
+    df = df[df[platforms].any(axis=1)]
+    df = df[(df['sentiment'] >= minimum_sentiment) &
+            (df['sentiment'] <= maximum_sentiment)]
```

Comment:
This is repeated in multiple functions. Could you make it so it isn't? DRY

Reply:
I'm embarrassed I repeated this code so many times, will definitely put into a neat function!